### PR TITLE
use sysctl 1.0

### PR DESCRIPTION
**Pull request by dhohengassner**
this means to remove the recipe used until now and use the ressource from sysctl

most of the code was already written by symondsandson
https://github.com/symondsandson/chef-os-hardening.git

### CODE REVIEW

**Comment by webframp** [comment]
Looks great @dhohengassner 

I'm waiting on this one before applying this to our systems so feedback from @chris-rock or other maintainers would be much appreciated!

**Comment by chris-rock** [comment]
This is huge @dhohengassner Thank you. Once question that I have. Can you still override the the attributes?

**Comment by dhohengassner** [comment]
Thanks for the fast response @chris-rock 

Good point! At the moment we can set the flags to enable/disable features as before.
But the values for the ressource calls are fixed now.

I think it would be good to use the same attributes the old sysctl cookbook used to set the values.
I will adapt the PR.

**Comment by chris-rock** [comment]
Nice, thank you @dhohengassner That sounds like the perfect approach.

**Comment by artem-sidorenko** [comment]
@dhohengassner I‘m currently on the parental leave, I hope @chris-rock can review here

**Comment by dhohengassner** [comment]
@chris-rock I reverted my resource calls and copied now parts of the old sysctl recipe into this cookbook.
I do not revert the changes in the unit tests because I see we have now less lines and I think they are easier to understand.
But I do not mind if you want to use the old tests.

Please review these changes.

Thank you!

**Comment by tas50** [comment]
This is an important fix for Chef 14 compatibility and it would be great to get this merged in as soon as possible.

File: recipes/sysctl.rb [hunk 1]
```diff
@@ -125,8 +127,14 @@
 node.default['sysctl']['params']['fs']['suid_dumpable'] =
   node['os-hardening']['security']['kernel']['enable_core_dump'] ? 2 : 0
 
-# include sysctl recipe and set /etc/sysctl.d/99-chef-attributes.conf
-include_recipe 'sysctl::apply'
+if node.attribute?('sysctl') && node['sysctl'].attribute?('params')
+  coerce_attributes(node['sysctl']['params']).each do |x|
```
Comment (chris-rock) [inline recipes/sysctl.rb]: Nice!

**Pull request merged**